js/components/addEditBookmark.js

```diff
@@ -23,7 +23,17 @@ class AddEditBookmark extends ImmutableComponent {
     this.onClick = this.onClick.bind(this)
     this.onSave = this.onSave.bind(this)
     this.onRemoveBookmark = this.onRemoveBookmark.bind(this)
+    this.state = {
+      bookmarkNameValid: true,
+      saveButtonDisabled: true
+    };
```

Comment:
The only reason I did this is because if I use only one porperty to check for both the button and the input, is that the input will be highlighted in read intially, which I think should happen only if you submit empty string. However with the disabling of the button if it is empty it is not necessary, so I will refactor.

Thanks for the feedback.